MSAL/src/instance/MSALDeviceInfoProvider.m

```diff
@@ -41,10 +41,27 @@ @implementation MSALDeviceInfoProvider
 - (void)deviceInfoWithRequestParameters:(MSIDRequestParameters *)requestParameters
                         completionBlock:(MSALDeviceInformationCompletionBlock)completionBlock
 {
+    void (^deviceInfoCompletion)(MSALDeviceInformation *, NSError *) = ^void(MSALDeviceInformation *msalDeviceInfo, NSError *error)
+    {
+        if (!msalDeviceInfo)
+        {
+            msalDeviceInfo = [MSALDeviceInformation new];
+            msalDeviceInfo.deviceMode = MSALDeviceModeDefault;
+        }
+
+        NSDictionary *deviceRegMetaDataInfo = [MSIDWorkPlaceJoinUtil getRegisteredDeviceMetadataInformation:nil];
+        if (deviceRegMetaDataInfo)
+        {
+            [msalDeviceInfo addRegisteredDeviceMetadataInformation:deviceRegMetaDataInfo];
+        }
+    
+        completionBlock(msalDeviceInfo, error);
+    };
+    
     if (![requestParameters shouldUseBroker])
     {
         NSError *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorBrokerNotAvailable, @"Broker is not enabled for this operation. Please make sure you have enabled broker support for your application", nil, nil, nil, nil, nil, YES);
-        completionBlock(nil, error);
+        deviceInfoCompletion(nil, error);
```

Comment:
If you're not using this error info, no need to create it. You can just switch it to logging.

Reply:
Yes we are not using ErrorInfo and we can remove and rely on logs but it will break other clients if they are using it.

Mostly we follows the below coding practice

if (error) {
 / /Handle Error
} else {
 // Process result
}

from Defender, We don't check error and instead directly check into the Dictionary for registration details but other client must be using above practice as they are interested in DeviceMode and ssoStatus